### ISSUE TITLE
The logloss gives -nan when using gblinear

### ISSUE THREAD

**Issue by supernovaf117** [dmlc/xgboost#213]
When I used gblinear in xgboost with logistic regression loss function for binary classification, at the beginning everything is normal ( like [43]). After several rounds the logloss becomes nan but the auc is still increasing ( like [44], [56] below).  I wonder what makes it nan.  The input data have been standardized by StandardScaler in scikit-learn.

[43]	train-logloss:0.331579	train-auc:0.874738	valid-logloss:0.332484	valid-auc:0.873435
boosting round 44, 91 sec elapsed
[44]	train-logloss:-nan	train-auc:0.874943	valid-logloss:0.332257	valid-auc:0.873648
boosting round 45, 93 sec elapsed
[45]	train-logloss:-nan	train-auc:0.875109	valid-logloss:0.332024	valid-auc:0.873822
boosting round 46, 95 sec elapsed
[46]	train-logloss:-nan	train-auc:0.875278	valid-logloss:0.331800	valid-auc:0.873999
boosting round 47, 97 sec elapsed
[47]	train-logloss:-nan	train-auc:0.875449	valid-logloss:0.331584	valid-auc:0.874177
boosting round 48, 99 sec elapsed
[48]	train-logloss:-nan	train-auc:0.875618	valid-logloss:0.331376	valid-auc:0.874353
boosting round 49, 100 sec elapsed
[49]	train-logloss:-nan	train-auc:0.875774	valid-logloss:0.331172	valid-auc:0.874516
boosting round 50, 102 sec elapsed
[50]	train-logloss:-nan	train-auc:0.875924	valid-logloss:0.330973	valid-auc:0.874674
boosting round 51, 104 sec elapsed
[51]	train-logloss:-nan	train-auc:0.876065	valid-logloss:0.330776	valid-auc:0.874821
boosting round 52, 106 sec elapsed
[52]	train-logloss:-nan	train-auc:0.876212	valid-logloss:0.330589	valid-auc:0.874975
boosting round 53, 108 sec elapsed
[53]	train-logloss:-nan	train-auc:0.876352	valid-logloss:0.330405	valid-auc:0.875122
boosting round 54, 110 sec elapsed
[54]	train-logloss:-nan	train-auc:0.876486	valid-logloss:0.330226	valid-auc:0.875262
boosting round 55, 112 sec elapsed
[55]	train-logloss:-nan	train-auc:0.876616	valid-logloss:0.330051	valid-auc:0.875398
boosting round 56, 114 sec elapsed
[56]	train-logloss:-nan	train-auc:0.876742	valid-logloss:-nan	valid-auc:0.875531
boosting round 57, 116 sec elapsed
[57]	train-logloss:-nan	train-auc:0.876866	valid-logloss:-nan	valid-auc:0.875661
boosting round 58, 118 sec elapsed

**Comment by tqchen** [comment]
can you check the most recent version of xgboost? This should be overflow issue in logloss evaluator and we made a fix on this

**Comment by supernovaf117** [comment]
Hi tqchen,
This issue disappeared when updating xgboost to the latest version.
Great! xgboost is so fast and memory-efficient. And it saves me lots of time.
Thanks

**Comment by tqchen** [comment]
I am glad it works out for you

Thanks for using xgboost
